src/chips/sam4l/dma.rs

```diff
@@ -212,6 +216,25 @@ impl DMAChannel {
 
         volatile_store(&mut registers.interrupt_enable, 1 << 1);
     }
+
+    pub fn do_xfer_buf(&self, pid: usize, 
+                       buf: Option<&'static mut [u8]>,
+                       len: usize) {
+        let ptr = buf.unwrap();
```

Comment:
I agree it should probably be a &'static mut [u8]. I stuck with Option because I beat my head agains thte wall for half a day trying to get the lifetimes right; it seemed like references in closures were having lifetimes elevated to the surrounding block, which made it difficult to use map(). The challenge was because the correct call order is to set up the buffers, then enable. So you need to map each of the dma_ Options in spi_dma twice (unlike USART, in which since you're only writing you can put it all in one map).